fish-rust/src/env_dispatch.rs

```diff
@@ -0,0 +1,800 @@
+            FLOGF!(term_support, "True-color support: enabling for iTerm");
+        }
+    } else if term.starts_with("st-") {
+        supports_24bit = true;
+        FLOGF!(term_support, "True-color support: enabling for st");
+    } else if let Some(vte) = vars.get(L!("VTE_VERSION")).map(|v| v.as_string()) {
+        if fish_wcstoi(vte.chars()).unwrap_or(0) > 3600 {
+            supports_24bit = true;
+            FLOGF!(
+                term_support,
+                "True-color support: enabling for VTE version ",
+                vte
+            );
+        }
+    }
+
+    let support = if supports_256color {
+        ColorSupport::TERM_256COLOR
+    } else {
+        ColorSupport::NONE
+    } | if supports_24bit {
+        ColorSupport::TERM_24BIT
+    } else {
+        ColorSupport::NONE
+    };
+    unsafe {
+        crate::output::output_set_color_support(support.bits() as i32);
+    }
+}
+
+/// Try to initialize the terminfo/curses subsystem using our fallback terminal name. Do not set
+/// `$TERM` to our fallback. We're only doing this in the hope of getting a functional shell.
+/// If we launch an external command that uses `$TERM`, it should get the same value we were given,
+/// if any.
+fn initialize_curses_using_fallbacks(vars: &dyn Environment) {
+    // xterm-256color is the most used terminal type by a massive margin, especially counting
+    // terminals that are mostly compatbile.
+    const FALLBACKS: [&str; 4] = ["xterm-256color", "xterm", "ansi", "dumb"];
+
+    let termstr = vars
+        .get_unless_empty(L!("TERM"))
+        .map(|v| v.as_string().to_string())
+        .unwrap_or(String::new());
+
+    for term in FALLBACKS {
+        // If $TERM is already set to the fallback name we're about to use, there's no point in
+        // seeing if the fallback name can be used.
+        if termstr == term {
+            continue;
+        }
+
+        let success = curses::setup(Some(term), libc::STDOUT_FILENO);
+        if is_interactive_session() {
+            if success {
+                FLOGF!(warning, wgettext!("Using fallback terminal type: "), term);
+            } else {
+                FLOGF!(
+                    warning,
+                    wgettext!("Could not set up terminal using the fallback terminal type: "),
+                    term,
+                );
+            }
+        }
+
+        if success {
+            break;
+        }
+    }
+}
+
+/// Apply any platform-specific hacks to our `cur_term`
+fn apply_term_hacks(vars: &dyn Environment) {
+    // Midnight Commander tries to extract the last line of the prompt, and does so in a way that is
+    // broken if you do '\r' after it like we normally do.
+    // See https://midnight-commander.org/ticket/4258.
+    if vars.get(L!("MC_SID")).is_some() {
+        crate::ffi::screen_set_midnight_commander_hack();
+    }
+
+    // Be careful, variables like `enter_italics_mode` are #defined to dereference through
+    // `cur_term`.
```

Comment:
Is this comment still accurate?

Reply:
No, but trying to get the value of `ENTER_ITALICS_MODE` et. al. without curses having been initialized will panic. I should update that comment.